use-shopping-cart/core/index.d.ts

```diff
@@ -0,0 +1,238 @@
+     * @param count The quantity to increase by
+     */
+    incrementItem: (id: string, count?: number) => void
+    /**
+     * Set the quantity of an item in the cart to a specific value
+     * @param id The ID of the item to change in quantity
+     * @param quantity The quantity to set the item to
+     */
+    setItemQuantity: (id: string, quantity: number) => void
+    /**
+     * The total price of the items in the cart
+     */
+    readonly totalPrice: number
+    /**
+     * Currency formatted version of `totalPrice`
+     */
+    readonly formattedTotalPrice: string
+    /**
+     * The number of items in the cart
+     */
+    readonly cartCount: number
+    /**
+     * Cart details is an object with IDs of the items in the cart as
+     * keys and details of the items as the value.
+     */
+    readonly cartDetails: CartDetails
+    /**
+     * Redirects customers to the Stripe checkout
+     * @param options {{ sessionId: string }} only used in CheckoutSession mode
+     * @returns Nothing or an error wrapped in a promise if an error occurred
+     */
+    redirectToCheckout: (options?: {
+      sessionId: string
+    }) => Promise<undefined | Error>
+    /**
+     * Redirects customers to the Stripe checkout
+     * @param options {{ id: string, quantity?: number }} only used in CheckoutSession mode
+     * @returns Nothing or an error wrapped in a promise if an error occurred
+     */
+    checkoutSingleItem: (options: {
+      id: string
+      quantity?: number
+    }) => Promise<undefined | Error>
+    /**
+     * Totally clears the cart of all items
+     */
+    clearCart: () => void
+    /**
+     * @param cartDetails cartDetails: [product.id]: {...}
+     * @param shouldMerge Boolean, defaults to true. If false, it would replace cartDetails
+     */
+    loadCart: (cartDetails: CartDetails, shouldMerge?: Boolean) => void
+  }
+
+  /**
+   * Provides several utilities and pieces of data for you to use in your application.
+   */
+  export function useShoppingCart(): ShoppingCartUtilities
+
+  /**
+   * Displays the values returned by `useShoppingCart()` in a table format.
+   */
+  export const DebugCart: React.FunctionComponent<React.DetailedHTMLProps<
+    React.TableHTMLAttributes<HTMLTableElement>,
+    HTMLTableElement
+  >>
+
+  interface FormatCurrencyStringProps {
+    /**
+     * The value to convert (i.e. 2599 in USD is $25.99)
+     */
+    value: number
+    /**
+     * The currency format (i.e. USD, CAD, GBP)
+     */
+    currency: string
+    /**
+     * The language to be used (i.e. en-US, fr-BE)
+     */
+    language?: string
+  }
+
+  /**
+   * Formats the value to a currency string
+   */
+  export function formatCurrencyString(props: FormatCurrencyStringProps): string
+
+  /**
+   * Utitlity for validating that the prices are what they should be server-side
+   * */
+  export function validateCartItems(
+    inventorySrc: [Product],
+    cartDetails: CartDetails
+  )
```

Comment:
Thor actually took care of this, just updated to make a copy

Reply:
I'm not sure I follow. This is along the same lines as the two sections of code that belong in `react.d.ts`: https://github.com/dayhaysoos/use-shopping-cart/pull/200#discussion_r632993862 and https://github.com/dayhaysoos/use-shopping-cart/pull/200#discussion_r632991733. But instead of `react.d.ts`, this needs its own `serverless.d.ts`.